amp-request-signer/src/main/java/org/apache/flink/connector/prometheus/sink/aws/AWS4SignerBase.java

```diff
@@ -0,0 +1,288 @@
+        String path = endpoint.getPath();
+        if (path == null || path.isEmpty()) {
+            return "/";
+        }
+
+        String encodedPath = urlEncode(path, true);
+        if (encodedPath.startsWith("/")) {
+            return encodedPath;
+        } else {
+            return "/".concat(encodedPath);
+        }
+    }
+
+    /**
+     * Examines the specified query string parameters and returns a canonicalized form.
+     *
+     * <p>The canonicalized query string is formed by first sorting all the query string parameters,
+     * then URI encoding both the key and value and then joining them, in order, separating key
+     * value pairs with an '&'.
+     *
+     * @param parameters The query string parameters to be canonicalized.
+     * @return A canonicalized form for the specified query string parameters.
+     */
+    public static String getCanonicalizedQueryString(Map<String, String> parameters) {
+        if (parameters == null || parameters.isEmpty()) {
+            return "";
+        }
+
+        SortedMap<String, String> sorted = new TreeMap<String, String>();
+
+        Iterator<Map.Entry<String, String>> pairs = parameters.entrySet().iterator();
+        while (pairs.hasNext()) {
+            Map.Entry<String, String> pair = pairs.next();
+            String key = pair.getKey();
+            String value = pair.getValue();
+            sorted.put(urlEncode(key, false), urlEncode(value, false));
+        }
+
+        StringBuilder builder = new StringBuilder();
+        pairs = sorted.entrySet().iterator();
+        while (pairs.hasNext()) {
+            Map.Entry<String, String> pair = pairs.next();
+            builder.append(pair.getKey());
+            builder.append("=");
+            builder.append(pair.getValue());
+            if (pairs.hasNext()) {
+                builder.append("&");
+            }
+        }
+
+        return builder.toString();
+    }
+
+    protected static String getStringToSign(
+            String scheme,
+            String algorithm,
+            String dateTime,
+            String scope,
+            String canonicalRequest) {
+        String stringToSign =
+                scheme
+                        + "-"
+                        + algorithm
+                        + "\n"
+                        + dateTime
+                        + "\n"
+                        + scope
+                        + "\n"
+                        + BinaryUtils.toHex(hash(canonicalRequest));
+        return stringToSign;
+    }
+
+    /** Hashes the string contents (assumed to be UTF-8) using the SHA-256 algorithm. */
+    public static byte[] hash(String text) {
+        try {
+            MessageDigest md = MessageDigest.getInstance("SHA-256");
+            md.update(text.getBytes("UTF-8"));
+            return md.digest();
+        } catch (Exception e) {
+            throw new RuntimeException(
+                    "Unable to compute hash while signing request: " + e.getMessage(), e);
```

Comment:
Can you replace this with a more specific exception? Same comment to all other `RuntimeException` usages